SourceX/qol.cpp

```diff
@@ -155,39 +190,103 @@ void DrawXPBar(CelOutputBuffer out)
 	if (!sgOptions.Gameplay.bExperienceBar)
 		return;
 
-	int barWidth = 306;
-	int barHeight = 5;
-	int yPos = gnScreenHeight - 9;                 // y position of xp bar
-	int xPos = (gnScreenWidth - barWidth) / 2 + 5; // x position of xp bar
-	int dividerHeight = 3;
-	int numDividers = 10;
-	int barColor = 198;
-	int emptyBarColor = 0;
-	int frameColor = 196;
-	bool space = true; // add 1 pixel separator on top/bottom of the bar
-
-	PrintGameStr(out, xPos - 22, yPos + 6, "XP", COL_WHITE);
-	int charLevel = plr[myplr]._pLevel;
-	if (charLevel == MAXCHARLEVEL - 1)
+	const PlayerStruct& player = plr[myplr];

```

Comment:
Finally! I was hoping that someone will start replacing all these `plr[whatever]` with references.

Reply:
Slowly but surely it's happening.

Reply:
> Finally! I was hoping that someone will start replacing all these plr[whatever] with references.

You are a few months late it seems @Iniquitatis 🤣